src/device.rs

```diff
@@ -42,3 +49,127 @@ impl Drop for Device {
+            FrequencyLevel::Simd256 => write!(f, "simd256"),
+            FrequencyLevel::Simd512 => write!(f, "simd512"),
+        }
+    }
+}
+
+/// Embree device configuration.
+pub struct Config {
+    /// The number of build threads. A value of 0 enables all detected hardware
+    /// threads. By default all hardware threads are used.
+    pub threads: u32,
+
+    /// The number of user threads that can be used to join and participate in a
+    /// scene commit using `rtcJoinCommitScene`.
+    pub user_threads: u32,
+
+    /// Whether build threads are affinitized to hardware threads. This is disabled
+    /// by default on standard CPUs, and enabled by default on Xeon Phi Processors.
+    pub set_affinity: bool,
+
+    /// When enabled, the build threads are started upfront. Useful for benchmarking
+    /// to exclude thread creation time. This is disabled by default.
+    pub start_threads: bool,
+
+    /// ISA selection. By default the ISA is chosen automatically.
+    pub isa: Isa,
+
+    /// Configures the automated ISA selection to use maximally the specified ISA.
+    pub max_isa: Isa,
+
+    /// Enables or disables usage of huge pages. Enabled by default under Linux but
+    /// disabled by default on Windows and macOS.
+    pub hugepages: bool,
+
+    /// Enables or disables the SeLockMemoryPrivilege privilege which is required to
+    /// use huge pages on Windows. This option has only effect on Windows and is ignored
+    /// on other platforms.
+    pub enable_selockmemoryprivilege: bool,
+
+    /// Verbosity of the output [0, 1, 2, 3]. No output when set to 0. The higher the
+    /// level, the more the output. By default the output is set to 0.
+    pub verbose: u32,
+
+    /// Frequency level the application want to run on. See [`FrequencyLevel`].
+    /// When some frequency level is specified, Embree will avoid doing optimizations
+    /// that may reduce the frequency level below the level specified.
+    pub frequency_level: FrequencyLevel,
+}
+
+impl Config {
+    /// Converts the configuration to a C string.
+    pub fn to_c_string(&self) -> CString {
+        let formated = format!("threads={},verbose={},set_affinity={},start_threads={},\
+        isa={},max_isa={},hugepages={},enable_selockmemoryprivilege={},frequency_level={}",
+                               self.threads, self.verbose, self.set_affinity as u32,
+                               self.start_threads as u32, self.isa, self.max_isa,
+                               self.hugepages as u32, self.enable_selockmemoryprivilege as u32,
+                               self.frequency_level).into_bytes();
+        unsafe {
+            CString::from_vec_unchecked(formated)
+        }
+    }
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Config {
+            threads: 0,
+            user_threads: 0,
+            set_affinity: false,
+            start_threads: false,
+            isa: Isa::Sse2,
```

Comment:
I think for ISA we should have a mode that is `None`, ie. let Embree pick the ISA based on the max ISA and the target machine. By defaulting ISA to SSE2 here and always including it in the generated Config string, I think we'll be forcing a default ISA of SSE2. So this could be an `Option<Isa>` instead.